src/main/java/com/github/onsdigital/babbage/api/util/LegacyCacheProxyMaxAge.java

```diff
@@ -0,0 +1,64 @@
+package com.github.onsdigital.babbage.api.util;
+
+import java.io.IOException;
+
+import org.apache.http.Header;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.client.methods.RequestBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+
+import com.github.onsdigital.logging.v2.event.SimpleEvent;
+
+public class LegacyCacheProxyMaxAge {
+
+    private final String legacyCacheProxyUrl;
+
+    public LegacyCacheProxyMaxAge(String legacyCacheProxyUrl) {
+        this.legacyCacheProxyUrl = legacyCacheProxyUrl;
+    }
+
+    public int getMaxAgeValueFromLegacyCacheProxy(String resourceURI) {
+        RequestBuilder requestBuilder = buildRequest(resourceURI);          
+        HttpUriRequest forwardRequest = requestBuilder.build();
+
+        int maxAgeValue = 0;
+        try {
+            try (CloseableHttpClient httpClient = HttpClients.createDefault()){
+                CloseableHttpResponse proxyResponse = httpClient.execute(forwardRequest);
+                maxAgeValue = extractMaxAgeValue(proxyResponse);
+            }
+        } catch (IOException e) {
+            SimpleEvent.error().exception(e).log("api MaxAge: error getting max age from legacy proxy");
+        }
+        return maxAgeValue;    
+    }
+    
+    private RequestBuilder buildRequest(String resourceURI) {
+        RequestBuilder requestBuilder = RequestBuilder.create("GET")
+                .setUri(legacyCacheProxyUrl);
+
+        requestBuilder.addHeader("uri", resourceURI);
```

Comment:
Hmmm... I don't understand how this works. Could you explain this, please? What is this uri header? I would have expected that all we'd need to do is to concatenate the legacy cache proxy URL with the resource URI. I don't get how this current implementation works.